### DISCUSSION TITLE
邮件通知设置教程

### DISCUSSION THREAD

**Discussion by Yink-Design** [walinejs/waline#192] [Q&A] [answered]
## 请问想要设置邮件通知配置变量具体要怎么做？

看文档不太懂，只需要在Vercel中添加环境变量吗？还是Leancloud也需要？我刚刚在Vercel中添加了但是没有用似乎

**Comment by lizheming** [comment, marked as answer]
评论通知只需要在 Vercel 中设置环境变量即可，注意设置完环境变量后需要重启服务才能生效环境变量。邮件通知这块基本就是配置的一些问题，在 Vercel 的日志中有详细的错误信息打印，可以看一下。另外也可以多试试其它几个种类的邮箱。

**Reply by Yink-Design** [reply]
感谢！还有如果想自定义邮件回复内容该如何修改？直接编辑环境变量内容吗？语法要求有哪些？

**Reply by lizheming** [reply]
`MAIL_SUBJECT_ADMIN` 和 `MAIL_TEMPLATE_ADMIN` 用于设置通知模板，见文档 https://waline.js.org/server/notification.html

模板可参考官方模板，模板语法是基于 nunjucks 的，具体规则就不解释了自己照猫画虎一下吧：https://github.com/lizheming/waline/blob/master/packages/server/src/service/notify.js#L196-L207

也有热心网友提供了一款模板主题，你也可以参考一下 https://github.com/walinejs/awesome-waline/tree/master/MAIL_TEMPLATE/style%201

**Reply by Yink-Design** [reply]
### 再次感谢！

**我改用QQ邮箱配置成功了** ，第一次使用Outlook365邮箱配置不成功，已经正确输入了SMTP服务器和端口以及账号密码等信息，但是还是无法链接。

查阅微软手册得知SMTP设置：

- 服务器名称: smtp.office365.com
- 端口: 587
- 加密方法: STARTTLS

但是无法正确使用邮件提醒，使用[nodemailer](https://github.com/nodemailer/nodemailer/blob/master/lib/well-known/services.json)配置 `SMTP_SERVICE` 变量后不行，于是直接手动配置了`SMTP_HOST` 和 `SMTP_PORT` 但还是无法链接。

是因为没有提供正确加密方法的原因吗？

**Reply by lizheming** [reply]
我没有用过 Outlook 邮箱，这块不太清楚了。tg 群里之前有用户用的 Outlook 说是可以的，不过速度比较慢超时比较严重。如果还是想配置 Outlook 的话可以加 tg 群里问一下吧。

**Reply by suyinLoong** [reply]
> ### 再次感谢！
> **我改用QQ邮箱配置成功了** ，第一次使用Outlook365邮箱配置不成功，已经正确输入了SMTP服务器和端口以及账号密码等信息，但是还是无法链接。
> 
> 查阅微软手册得知SMTP设置：
> 
> * 服务器名称: smtp.office365.com
> * 端口: 587
> * 加密方法: STARTTLS
> 
> 但是无法正确使用邮件提醒，使用[nodemailer](https://github.com/nodemailer/nodemailer/blob/master/lib/well-known/services.json)配置 `SMTP_SERVICE` 变量后不行，于是直接手动配置了`SMTP_HOST` 和 `SMTP_PORT` 但还是无法链接。
> 
> 是因为没有提供正确加密方法的原因吗？

我用qq邮箱配置的邮件通知，但是不生效，查看日志也没用报错。但是配置的使用qq通知就生效了，不知道怎么回事

**Reply by suyinLoong** [reply]
MAIL_SUBJECT：评论回复邮件标题自定义
MAIL_TEMPLATE：评论回复邮件内容自定义
MAIL_SUBJECT_ADMIN：新评论通知邮件标题自定义
MAIL_TEMPLATE_ADMIN：新评论通知邮件内容自定义

这几个选项是必填吗？

**Reply by lizheming** [reply]
邮件通知分为新评论通知和评论回复通知，新评论通知如果已经开启了 QQ/Wechat/Telegram 通知的话不会发送邮件通知。看看是不是这种情况，如果不是这种情况的话请自行检查一下邮箱相关的配置。另外模板相关的配置都是选填的，会有默认模板。

**Reply by ChrisGray0626** [reply]
> 评论通知只需要在 Vercel 中设置环境变量即可，注意设置完环境变量后需要重启服务才能生效环境变量。邮件通知这块基本就是配置的一些问题，在 Vercel 的日志中有详细的错误信息打印，可以看一下。另外也可以多试试其它几个种类的邮箱。

您好我GMAIL也始终没有配置成功，请问是有什么特殊设置要求吗？

**Reply by cao-gift** [reply]
> 邮件通知分为新评论通知和评论回复通知，新评论通知如果已经开启了 QQ/Wechat/Telegram 通知的话不会发送邮件通知。看看是不是这种情况，如果不是这种情况的话请自行检查一下邮箱相关的配置。另外模板相关的配置都是选填的，会有默认模板。

可不可以新评论通知已经开启了QQ/Wechat/Telegram 通知的情况下，也可以发送邮件通知

**Reply by lizheming** [reply]
可以发送回复邮件通知，不支持此时再发送新评论邮件通知

**Comment by suyinLoong** [comment]
**经测试，我的qq邮箱邮箱通知直接就不生效，不是因为开通了QQ通知的原因**
**QQ邮箱：**
    - `SMTP_SERVICE`的值为`QQ`就行吧，然后`SMTP_HOST`和`SMTP_PORT`的值是不是就不用填写了？
    - `SMTP_PASS`是填写登录密码？还是独立密码？又或者是授权码？
**163邮箱：**
    - `SMTP_PASS`这里填写的授权码配置成功了
    - `SMTP_SERVICE`的值为`163`

**Reply by suyinLoong** [reply]
> **经测试，我的qq邮箱邮箱通知直接就不生效，不是因为开通了QQ通知的原因**
> **QQ邮箱：**
> - `SMTP_SERVICE`的值为`QQ`就行吧，然后`SMTP_HOST`和`SMTP_PORT`的值是不是就不用填写了？
> - `SMTP_PASS`是填写登录密码？还是独立密码？又或者是授权码？
> **163邮箱：**
> - `SMTP_PASS`这里填写的授权码配置成功了
> - `SMTP_SERVICE`的值为`163`

经测试，QQ邮箱的`SMTP_PASS`也需要填写成授权码才行！

**Reply by magic-H728** [reply]
我尝试了163和QQ邮箱都失败了唉

**Reply by gumpBOBO** [reply]
> **经测试，我的qq邮箱邮箱通知直接就不生效，不是因为开通了QQ通知的原因** **QQ邮箱：** - `SMTP_SERVICE`的值为`QQ`就行吧，然后`SMTP_HOST`和`SMTP_PORT`的值是不是就不用填写了？ - `SMTP_PASS`是填写登录密码？还是独立密码？又或者是授权码？ **163邮箱：** - `SMTP_PASS`这里填写的授权码配置成功了 - `SMTP_SERVICE`的值为`163`


SMTP_SERVICE  qq
SMTP_USER       ??@qq.com
SITE_NAME        博客名称
AUTHOR_EMAIL ??@qq.com
SMTP_PASS        qq设置通过手机获取的验证码
SENDER_EMAIL  ??@qq.com(和下面的变量必须同时配，通知才会更改，不然显示的邮箱)
SENDER_NAME  博客名称(邮件通知名称)
SITE_URL             https://xxx (一定要+https://)

模版的环境变量跟是否成功没太大关系，你先把默认通知接到在改模版

**Comment by hiifong** [comment]
邮件通知模板你们是怎么写的

**Reply by musnows** [reply]
> 邮件通知模板你们是怎么写的

https://www.sarakale.top/blog/posts/537344b2.html#post-comment 这里面有好几个不错的模板，可以看看

**Reply by hiifong** [reply]
谢谢大佬

**Comment by kdyzm** [comment]
我配置的环境变量SENDER_NAME没生效，你们知道咋回事吗

---
The environment variable SENDER_NAME I configured did not take effect. Do you know what happened?

**Comment by musnows** [comment]
> 我配置的环境变量SENDER_NAME没生效，你们知道咋回事吗
> 
> The environment variable SENDER_NAME I configured did not take effect. Do you know what happened?

你要看看你是什么邮箱平台，163邮箱我这边测试是可以修改成功的。但是163的SENDER_EMAIL必须和AUTHOR_EMAIL保持一致。
